File: src/main/java/core/algorithm/SimpleAlg.java
package core.algorithm;

import core.data.Data;

import java.util.function.Function;

/**
 * Simple class to wrap methods into Algorithm interface in one line of code
 * @param <I>
 * @param <O>
 */
public class SimpleAlg<I extends Data, O extends Data> implements Algorithm<I, O> {

    private Function<I, O> func;

    public SimpleAlg(Function<I, O> func){
        this.func = func;
    }

    @Override
    public O execute(I input) {
        return func.apply(input);
    }
}

File: src/main/kotlin/com/mirf/core/algorithm/SimpleAlg.kt
package com.mirf.core.algorithm

import com.mirf.core.data.Data

/**
 * Simple class to wrap methods into Algorithm interface in one line of code
 * @param <I>
 * @param <O>
</O></I> */
class SimpleAlg<I : Data, O : Data>(private val func: (I) -> O) : Algorithm<I, O> {

    override fun execute(input: I): O {
        return func.invoke(input)
    }
}
